pkg/types/codec.go

```diff
@@ -1,3 +1,63 @@
+/*
+Codec is an interface that provides encoding and decoding functionality for a specific type identified by a name.
+Because there are many types that a ContractReader or ContractWriter can either accept or return, all encoding
+instructions provided by the codec are based on the type name.
+
+Starting from the lowest level, take for instance a big.Int encoder where we want the output to be big endian binary
+encoded.
+
+	typeCodec, _ := binary.NewBigInt(32, true, binary.BigEndian())
```

Comment:
I couldn't find `binary.NewBigInt`. It looks like this is supposed to be a simple example, but you don't show how to implement it. The godoc has a more complete `example_test.go`, you could reference this as `example_minimal_test.go`.